### PR TITLE
Add is_sent_end to token attributes

### PR DESCRIPTION
**Pull request by MiniLau**
## Description

Added #3637.
I added some basic tests in the tokenizer and the senten 

### Types of change

Added a new feature.
Added test for Tokenizer and Sentencizer to test the new attribute

## Checklist

- [X] I have submitted the spaCy Contributor Agreement.
- [X] I ran the tests, and all new and existing tests passed.
- [X] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy/symbols.pxd [hunk 1]
```diff
@@ -83,6 +83,7 @@ cdef enum symbol_t:
     ENT_TYPE
     HEAD
     SENT_START
+    SENT_END
```
Comment (honnibal) [inline spacy/symbols.pxd]: Unfortunately the order of the symbols and attributes matters -- it influences which ID will be assigned to that attribute. So we need new attributes to go to the end, and we can only reorder them when we're ready to break model compatibility.

If we insert the `SENT_END` symbol here, the subsequent symbols will be mapped to different values, which could break models.

File: spacy/structs.pxd [hunk 2]
```diff
@@ -75,6 +75,7 @@ cdef struct TokenC:
     uint32_t r_edge
 
     int sent_start
+    int sent_end
```
Comment (honnibal) [inline spacy/structs.pxd]: We shouldn't add this to the struct, that's a waste of space and it's data duplication that's hard to keep in-sync with the `sent_start` attribute. It's okay to have `sent_end` as an attribute, but it has to be computed dynamically by checking whether the following token is `sent_start`.

File: spacy/attrs.pxd [hunk 3]
```diff
@@ -86,6 +86,7 @@ cdef enum attr_id_t:
     ENT_TYPE
     HEAD
     SENT_START
+    SENT_END
```
Comment (honnibal) [inline spacy/attrs.pxd]: We need to place `SENT_END` at the end of the list for now -- see the comment on the `symbols.pyx`.

**Comment by MiniLau** [comment]
Thank you for the implementation approval. So, to finalize the pull request I should just edit the 3 files the way you explained above ?

**Comment by honnibal** [comment]
The changes would need to be a bit bigger than those three files. You'll also need to change the implementation of the `sent_end` property in `token.pyx`, because you'll need to compute it based on the following token's `sent_start`.

**Comment by MiniLau** [comment]
I don't understand, I thought the implementation of the property was right. Why should I change it ? I can't use "sent_start" property in the implementation in token.pyx ?

**Comment by honnibal** [comment]
@MiniLau My mistake, sorry --- yes the property's implementation is correct.

**Comment by MiniLau** [comment]
Is is okay with those modifications ?

**Comment by adrianeboyd** [comment]
Sorry, I think this got overlooked because it looked like there will still outstanding changes requested. Let me see if I can resolve the conflicts and let the CI run the tests again.

**Comment by adrianeboyd** [comment]
@honnibal: Can you have another look at this?

There are still some minor changes to make to the `Matcher` to make `IS_SENT_END` work like `IS_SENT_START`, but I could do them in a separate PR.

**Comment by adrianeboyd** [comment]
Since the original repo is no longer available, most of this PR is reconstructed in #5375.

**Pull request closed without merging**